contents/docs/integrate/send-events/_snippets/send-events-ios.mdx

```diff
@@ -34,4 +34,30 @@ If you want to manually send a new screen capture event, use the `screen` functi
 
 ```swift
 PostHogSDK.shared.screen("Dashboard", properties: ["fromIcon": "bottom"])
+```
+
+### Capturing screen views in SwiftUI
+
+In SwiftUI, views can exist at multiple levels of abstraction, making automatic tracking challenging. A SwiftUI view might represent an entire screen, similar to a `UIViewController`, but it could also represent much smaller UI components, like `UIControl` or `UIView` elements. Because SwiftUI doesn’t clearly distinguish between these levels of views in the same way UIKit does, it becomes difficult to reliably auto-track "screen" (as in full-screen) views.
+
+Moreover, because of SwiftUI's declarative nature and dynamic view construction, SwiftUI views may sometimes lack clear, static names that are suitable for analytics purposes. 
+
+To track a screen view, you can explicitly apply the `postHogScreenView` view modifier to your full-screen view. While the PostHog iOS SDK will attempt to infer the screen's name based on the view's `Type`, you can also pass a custom screen name that better describes your view, along with optional event properties.
+
+```swift 
+// This will trigger a screen view event with $screen_name: "HomeView"
+struct HomeView: View {
+    var body: some View {
+        Text("Welcome to MyCoolView")
+            .postHogScreenView()
```

Comment:
This is in the`Text` component, are you sure it's inferring `HomeView`?

Reply:
This section is about the screen view, the function is also about the screen view, so setting the modifier to a `Text` would not make sense.
I didn't find a way to not allow the modifier to non root views because everything is a `View` anyway.

Reply:
the code snippet above makes more sense, yes

Reply:
this part of the code snippet: `.postHogScreenView("My Home View"` is wrong though, since this part of the docs is the inferred screen name, so you cannot pass the string `"My Home View"` here, you just need to be sure that the inferred name is either `HomeViewContent` or `HomeView`, and fix the comments